### ISSUE TITLE
'\0' as package delimiter

### ISSUE THREAD

**Issue by aqrln** [metarhia/JSTP#11]
Wouldn't it be better to use null characters as package delimiters? It is only one byte instead of three and it will under no circumstances appear in text data.

**Comment by tshemsedinov** [comment]
Strings may contain escaped `\u0000` but this delimiter it is not valid JavaScript array element. Now we use buffers: `'[' + '{ valid: 'js object'},{\f},{ valid: 'js object'}' + ',{\f},' + '{ valid: 'js object'}' + ']'` and we can concatenate buffers and parse. But `'[' + '{ valid: 'js object'},□,{ valid: 'js object'}' + '□' + '{ valid: 'js object'}' + ']'` where `□` is a null char delimiter can't be parsed. Maybe we can filter array of buffer to remove all delimiters before parsing. Please test performance.

**Comment by aqrln** [comment]
All null characters should be just converted to commas when a packet arrives. As a side-effect benefit, we will no longer store empty objects in buffer and there will be no need to drop every odd-indexed entry in the buffer.

**Comment by tshemsedinov** [comment]
Agreed

**Comment by aqrln** [comment]
This will be implemented during JSTP standalone library rewrite, which is happening right in this repo. Stay tuned and prepare your implementations @NechaiDO @belochub @lundibundi @lidaamber @Gagnant @GYFK ;)

**Comment by aqrln** [comment]
Also, a few notes about WebSocket transport.

1. We don't need any package delimiters for WebSocket. The WebSocket API used in browsers doesn't expose the vanilla frame-based or streaming API (though supported by WebSocket protocol) but only message-based one which splits messages into frames and aggregates them back together automatically, and all the major WebSocket implementations are capable of that too, so there's no need to build the same mechanism again on top of WebSocket and induce unnecessary overhead because of situation that will never happen.

2. Even if we wanted to (say, for simplicity, like it is actually implemented in Impress currently), we couldn't do it easily with the new separator. WebSocket frames are either text (UTF8-encoded) or binary. I doubt that we can send a null byte in a text packet (TODO: check it) and binary messages are very inconvenient to process in the browser environment since it lacks something like Node.js Buffer while supported ArrayBuffer is kinda shitty.

TL;DR: rewrite that quick and dirty WebSocket connection adapter, really decouple transports, don't use packet delimiters for WS, only for TCP connections.
